File: Perfect Number/perfect_number.py
import math
n = int(input("Enter a number: "))

divisors = {1}

for i in range(2, math.ceil(n ** .5) + 1):
    if n % i == 0:
        divisors.add(i)
        divisors.add(n // i)

if sum(divisors) == n:
    print("It is a Perfect Number")
else:
    print("NOT a Perfect Number")

File: Perfect Number/PerfectNumber.java
import java.io.*;
import java.util.*;

class Main
{
    static String perfectNumber(int n){
        HashSet<Integer> divisors = new HashSet<>();

        divisors.add(1);
        for (int i = 2; i * i < n + 1; i++){
            if (n % i == 0){
                divisors.add(i);
                divisors.add((int) (n / i));
            }
        }

        int sum = 0;
        for (int i: divisors){
            sum += i;
        }

        return sum == n ? "Is a Perfect Number" : "Not a Perfect Number";

    }


    public static void main(String[] args){
        int n = 496;
        System.out.println(perfectNumber(n));
    }
}